storage/statedb/cache_redis.go

```diff
@@ -76,22 +86,45 @@ func NewRedisCache(config TrieNodeCacheConfig) (*RedisCache, error) {
 		return nil, err
 	}
 
-	logger.Info("Initialize trie node cache with redis", "endpoint", config.RedisEndpoints,
+	cache := &RedisCache{
+		client:    cli,
+		setItemCh: make(chan setItem, redisSetItemChannelSize),
+		pubSub:    cli.Subscribe(),
+	}
+
+	workerNum := runtime.NumCPU() / 2
```

Comment:
How about adding a guard for single-core machine just in case?

Reply:
Good suggestion. I will add 1. Thanks!